More detailed explanation of initial_fetch caching

PaulHobbs opened the pull request:
This CL adds some more explanation of how initial_fetch (the client-side "fetch" function in load) caches results *keyed on URL*.

I was really scratching my head when my client-side requests were returning bogus results, even when the server-side was working. It turns out, because I was making requests using parameters passed in the POST request body, semantically different requests to the same URL were getting deduped by svelte's client-side request caching. This documentation change adds a warning about this case, with a bit more detail about how initial_fetch works.

nitroboy720 reviewed (changes requested):
2 little typos in the first sentence: 
> Any client-side request to `the a` given URL will be intercepted and immediately return the same response that the server `recieved.`

Personal opinion: This first sentence could be removed altogether, since it doesn't add to what is said just above it.

PaulHobbs commented:
> Personal opinion: This first sentence could be removed altogether, since it doesn't add to what is said just above it.

Thanks for the feedback - I removed it. I also added a parenthetical "(keyed on URL)" to the previous sentence to make it more specific about how the caching works.

Rich-Harris commented:
Thank you — I'm going to close this, since the behaviour you describe is a bug, and not something people would ideally need to think about: #633
